matchtree.go

```diff
@@ -983,14 +991,33 @@ func (d *indexData) newMatchTree(q query.Q, opt matchTreeOpt) (matchTree, error)
 		optCopy := opt
 		optCopy.DisableWordMatchOptimization = true
 
-		subMT, err := d.newMatchTree(s.Expr, optCopy)
+		anchored := false
+		expr := s.Expr
+		switch e := s.Expr.(type) {
+		case *query.Regexp:
+			pattern := e.Regexp.String()
+			if strings.HasPrefix(pattern, "^") && strings.HasSuffix(pattern, "$") {
+				pattern = pattern[1 : len(pattern)-1]
+				parsedPattern, err := syntax.Parse(pattern, e.Regexp.Flags)
+				if err != nil {
+					return nil, err
+				}
```

Comment:
we shouldn't return an error here, we should fallback to the non-optimized path.

Reply:
using the AST, this becomes infallible, so this is fixed in the latest commit